src/analytics/dsl.tsx

```diff
@@ -0,0 +1,12 @@
+export const dsl = {
+  "Follow Artist": {
+    intent: (object) => ({ artist_id: object.artist.id }),
+    success: (object) => ({ artist_id: object.artist.id }),
+    fail: (object) => ({ artist_id: object.artist.id }),
```

Comment:
You could add an interface to specify the handlers signature
```
const dsl: [string: AnalyticsHandler] = {...}

interface AnalyticsHandler {
  intent: (object: any) => any,
  success: (object: any) => any,
  error: (object: any, error: Error) => any
}
```
or if we end up using the `Analytics.on` API style
```
class Analytics {
  on(contant: string, handler: AnalyticsHandler) {...}
}
```